mip/mip_db_data_extractors/dc_extractor.py

```diff
@@ -45,11 +71,42 @@ def __init__(self,
         self._candidates_tables = candidates_tables
         self._dc_candidates_sets = None
 
+    def _test_comparison_atoms(self) -> bool:
+        """Test the validity of a the comparison atoms.
+        """""
+        for comparison_atom in self._comparison_atoms:
+            input_type = utils.check_string_type(comparison_atom[0])
+            if input_type == 'name' and not self._test_variable_name(comparison_atom[0]):
+                return False
+            input_type = utils.check_string_type(comparison_atom[2])
+            if input_type == 'name' and not self._test_variable_name(comparison_atom[2]):
+                return False
+        return True
+
+    def _test_variable_name(self, var_name: str) -> bool:
+        """Given a variable name, test that it is defined in the dc dict.
+        """""
+        for ls in self._dc_dict.values():
+            for tp in ls:
+                if tp[0] == var_name:
+                    return True
+        return False
```

Comment:
not too critical but I think it's cleaner
```suggestion
    def _test_variable_name(self, var_name: str) -> bool:
        """Given a variable name, test that it is defined in the dc dict.
        """""
        return any(tp[0] == var_name for ls in self._dc_dict.values() for tp in ls)
```

Reply:
Done.